<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-GB"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Prepare Input Files for MStrat — prep_mstrat_input • avial</title><!-- mathjax math --><script src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script><script>
  window.MathJax = {
    chtml: {
      fontURL: "https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/output/chtml/fonts/woff-v2"
    }
  };
</script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Prepare Input Files for MStrat — prep_mstrat_input"><meta name="description" content="Prepare input files for MStrat, a software for building germplasm core
collections by maximizing allelic or phenotypic richness
(Schoen and Brown 1993; Gouesnard et al. 2001; Gouesnard et al. 2002)
."><meta property="og:description" content="Prepare input files for MStrat, a software for building germplasm core
collections by maximizing allelic or phenotypic richness
(Schoen and Brown 1993; Gouesnard et al. 2001; Gouesnard et al. 2002)
."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">avial</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fa-home fa-lg"></span> Home</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html"><span class="fa fa-code"></span> Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/Highlight-Checks.html">Highlight Genotypes in Group-wise Histograms using Group-wise Point and Error Bar Plots</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/aravind-j/avial"><span class="fa fab fa-github-alt"></span> Github</a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Prepare Input Files for <code>MStrat</code></h1>
      <small class="dont-index">Source: <a href="https://github.com/aravind-j/avial/blob/master/R/prep_mstrat_input.R" class="external-link"><code>R/prep_mstrat_input.R</code></a></small>
      <div class="d-none name"><code>prep_mstrat_input.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Prepare input files for <code>MStrat</code>, a software for building germplasm core
collections by maximizing allelic or phenotypic richness
(Schoen and Brown 1993; Gouesnard et al. 2001; Gouesnard et al. 2002)
.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">prep_mstrat_input</span><span class="op">(</span></span>
<span>  <span class="va">data</span>,</span>
<span>  <span class="va">genotype</span>,</span>
<span>  <span class="va">qualitative</span>,</span>
<span>  <span class="va">quantitative</span>,</span>
<span>  <span class="va">active</span>,</span>
<span>  <span class="va">target</span>,</span>
<span>  center <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  weights.qualitative <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  weights.quantitative <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  nclass.quantitative <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  always.selected <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  file.name <span class="op">=</span> <span class="st">"MStrat_input"</span>,</span>
<span>  folder.path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-data">data<a class="anchor" aria-label="anchor" href="#arg-data"></a></dt>
<dd><p>The data as a data frame object. The data frame should possess
columns with the genotype names and multiple quantitative and/or
qualitative trait/variable data.</p></dd>


<dt id="arg-genotype">genotype<a class="anchor" aria-label="anchor" href="#arg-genotype"></a></dt>
<dd><p>Name of column with the genotype names as a character string.</p></dd>


<dt id="arg-qualitative">qualitative<a class="anchor" aria-label="anchor" href="#arg-qualitative"></a></dt>
<dd><p>Name of columns with the qualitative traits as a character
vector.</p></dd>


<dt id="arg-quantitative">quantitative<a class="anchor" aria-label="anchor" href="#arg-quantitative"></a></dt>
<dd><p>Name of columns with the quantitative traits as a
character vector.</p></dd>


<dt id="arg-active">active<a class="anchor" aria-label="anchor" href="#arg-active"></a></dt>
<dd><p>Name of traits/variables to be declared as active.</p></dd>


<dt id="arg-target">target<a class="anchor" aria-label="anchor" href="#arg-target"></a></dt>
<dd><p>Name of traits/variables to be declared as target.</p></dd>


<dt id="arg-center">center<a class="anchor" aria-label="anchor" href="#arg-center"></a></dt>
<dd><p>either a logical value or numeric-alike vector of length
    equal to the number of columns of <code>x</code>, where
    ‘numeric-alike’ means that <code><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a>(.)</code> will
    be applied successfully if <code><a href="https://rdrr.io/r/base/numeric.html" class="external-link">is.numeric</a>(.)</code> is not true.</p></dd>


<dt id="arg-scale">scale<a class="anchor" aria-label="anchor" href="#arg-scale"></a></dt>
<dd><p>either a logical value or a numeric-alike vector of length
    equal to the number of columns of <code>x</code>.</p></dd>


<dt id="arg-weights-qualitative">weights.qualitative<a class="anchor" aria-label="anchor" href="#arg-weights-qualitative"></a></dt>
<dd><p>An vector of weight to be applied on a qualitative
traits. Should be <code>NULL</code> or a numeric vector of the same length as the
number of qualitative traits. If <code>NULL</code>, the default weight of 1 is
given.</p></dd>


<dt id="arg-weights-quantitative">weights.quantitative<a class="anchor" aria-label="anchor" href="#arg-weights-quantitative"></a></dt>
<dd><p>An vector of weight to be applied on a
quantitative traits. Should be <code>NULL</code> or a numeric vector of the same
length as the number of quantitative traits. If <code>NULL</code>, the default
weight of 1 is given.</p></dd>


<dt id="arg-nclass-quantitative">nclass.quantitative<a class="anchor" aria-label="anchor" href="#arg-nclass-quantitative"></a></dt>
<dd><p>The number of classes into which each quantitative
trait data have to be divided into. Should be <code>NULL</code> or a integer
vector of the same length as the number of quantitative traits. If
<code>NULL</code>, the default of 5 is applied. MStat limits the maximum number
of classes to 1000.</p></dd>


<dt id="arg-always-selected">always.selected<a class="anchor" aria-label="anchor" href="#arg-always-selected"></a></dt>
<dd><p>A character vector with names of individuals in the
<code>genotype</code> that should always be selected in the core collection. The
maximum length accepted by <code>MStrat</code> is 500.</p></dd>


<dt id="arg-file-name">file.name<a class="anchor" aria-label="anchor" href="#arg-file-name"></a></dt>
<dd><p>A character string of name of file where the data will be
saved.</p></dd>


<dt id="arg-folder-path">folder.path<a class="anchor" aria-label="anchor" href="#arg-folder-path"></a></dt>
<dd><p>The path to folder where the input files are to be saved.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="note">Note<a class="anchor" aria-label="anchor" href="#note"></a></h2>
    <p>Any <code>NA</code> values will be considered as missing value and converted
  to 9999 which is the code for the same in <code>MStrat</code>.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Gouesnard B, Bataillon TM, Decoux G, Rozale C, Schoen DJ, David JL (2001).
“MSTRAT: An algorithm for building germ plasm core collections by maximizing allelic or phenotypic richness.”
<em>Journal of Heredity</em>, <b>92</b>(1), 93–94.<br><br> Gouesnard B, Bataillon TM, Decoux G, Rozale C, Schoen DJ, David JL (2002).
“MStrat Documentation 1.1.”
Evolutionary genomics and population management (GE\(^{\textrm{2}}\)pop), Institut Amelioration Genetique et Adaptation des Plantes mediterraneennes et tropicales (agAp Institute), CIRAD, Montpellier, France.<br><br> Schoen DJ, Brown AH (1993).
“Conservation of allelic richness in wild crop relatives is aided by assessment of genetic markers.”
<em>Proceedings of the National Academy of Sciences</em>, <b>90</b>(22), 10623–10627.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/aravind-j/EvaluateCore" class="external-link">EvaluateCore</a></span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Registered S3 method overwritten by 'vegan':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   method     from      </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   rev.hclust dendextend</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --------------------------------------------------------------------------------</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Welcome to EvaluateCore version 0.1.4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # To know whats new in this version type:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   news(package='EvaluateCore')</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   for the NEWS file.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # To cite the methods in the package type:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   citation(package='EvaluateCore')</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> # To suppress this message use:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   suppressPackageStartupMessages(library(EvaluateCore))</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --------------------------------------------------------------------------------</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">cassava_EC</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">cassava_EC</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">quant</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"NMSR"</span>, <span class="st">"TTRN"</span>, <span class="st">"TFWSR"</span>, <span class="st">"TTSW"</span>, <span class="st">"TTPW"</span>, <span class="st">"AVPW"</span>,</span></span>
<span class="r-in"><span>           <span class="st">"ARSR"</span>, <span class="st">"SRDM"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">qual</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CUAL"</span>, <span class="st">"LNGS"</span>, <span class="st">"LFRT"</span>, <span class="st">"LBTEF"</span>, <span class="st">"CBTR"</span>, <span class="st">"NMLB"</span>,</span></span>
<span class="r-in"><span>          <span class="st">"ANGB"</span>, <span class="st">"CUAL9M"</span>, <span class="st">"LVC9M"</span>, <span class="st">"TNPR9M"</span>, <span class="st">"PL9M"</span>, <span class="st">"STRP"</span>, <span class="st">"STRC"</span>,</span></span>
<span class="r-in"><span>          <span class="st">"PSTR"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Prepare genotype column</span></span></span>
<span class="r-in"><span><span class="va">data</span><span class="op">$</span><span class="va">Accession</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span></span>
<span class="r-in"><span><span class="va">data</span><span class="op">$</span><span class="va">Accession</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">Accession</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Convert qualitative data as factors</span></span></span>
<span class="r-in"><span><span class="va">data</span><span class="op">[</span>, <span class="va">qual</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">data</span><span class="op">[</span>, <span class="va">qual</span><span class="op">]</span>,</span></span>
<span class="r-in"><span>                       <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">active</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"LNGS"</span>, <span class="st">"LFRT"</span>, <span class="st">"LBTEF"</span>, <span class="st">"CBTR"</span>, <span class="st">"NMLB"</span>,</span></span>
<span class="r-in"><span>           <span class="st">"ANGB"</span>, <span class="st">"CUAL9M"</span>, <span class="st">"LVC9M"</span>, <span class="st">"TNPR9M"</span>,</span></span>
<span class="r-in"><span>           <span class="st">"TTRN"</span>, <span class="st">"TFWSR"</span>, <span class="st">"TTSW"</span>, <span class="st">"TTPW"</span>, <span class="st">"AVPW"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">target</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"NMSR"</span>, <span class="st">"TTRN"</span>, <span class="st">"ARSR"</span>, <span class="st">"SRDM"</span>,</span></span>
<span class="r-in"><span>           <span class="st">"CUAL"</span>, <span class="st">"LNGS"</span>, <span class="st">"TNPR9M"</span>,</span></span>
<span class="r-in"><span>           <span class="st">"PL9M"</span>, <span class="st">"STRP"</span>, <span class="st">"STRC"</span>,</span></span>
<span class="r-in"><span>           <span class="st">"PSTR"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">sel</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TMe-2906"</span>, <span class="st">"TMe-3412"</span>, <span class="st">"TMe-1374"</span>, <span class="st">"TMe-768"</span>, <span class="st">"TMe-14"</span>,</span></span>
<span class="r-in"><span>         <span class="st">"TMe-3284"</span>, <span class="st">"TMe-937"</span>, <span class="st">"TMe-1859"</span>, <span class="st">"TMe-3265"</span>, <span class="st">"TMe-1739"</span>,</span></span>
<span class="r-in"><span>         <span class="st">"TMe-972"</span>, <span class="st">"TMe-769"</span>, <span class="st">"TMe-3243"</span>, <span class="st">"TMe-3719"</span>, <span class="st">"TMe-1095"</span>,</span></span>
<span class="r-in"><span>         <span class="st">"TMe-893"</span>, <span class="st">"TMe-1262"</span>, <span class="st">"TMe-2083"</span>, <span class="st">"TMe-376"</span>, <span class="st">"TMe-3633"</span>,</span></span>
<span class="r-in"><span>         <span class="st">"TMe-1738"</span>, <span class="st">"TMe-2428"</span>, <span class="st">"TMe-259"</span>, <span class="st">"TMe-3457"</span>, <span class="st">"TMe-1406"</span>,</span></span>
<span class="r-in"><span>         <span class="st">"TMe-977"</span>, <span class="st">"TMe-3006"</span>, <span class="st">"TMe-925"</span>, <span class="st">"TMe-3671"</span>, <span class="st">"TMe-2779"</span>,</span></span>
<span class="r-in"><span>         <span class="st">"TMe-1293"</span>, <span class="st">"TMe-268"</span>, <span class="st">"TMe-266"</span>, <span class="st">"TMe-3562"</span>, <span class="st">"TMe-801"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">prep_mstrat_input</span><span class="op">(</span>data <span class="op">=</span> <span class="va">data</span>, genotype <span class="op">=</span> <span class="st">"Accession"</span>,</span></span>
<span class="r-in"><span>                  qualitative <span class="op">=</span> <span class="va">qual</span>, quantitative <span class="op">=</span> <span class="va">quant</span>,</span></span>
<span class="r-in"><span>                  active <span class="op">=</span> <span class="va">active</span>, target <span class="op">=</span> <span class="va">target</span>,</span></span>
<span class="r-in"><span>                  center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>                  weights.qualitative <span class="op">=</span> <span class="cn">NULL</span>,</span></span>
<span class="r-in"><span>                  weights.quantitative <span class="op">=</span> <span class="cn">NULL</span>,</span></span>
<span class="r-in"><span>                  nclass.quantitative <span class="op">=</span> <span class="cn">NULL</span>, always.selected <span class="op">=</span> <span class="va">sel</span>,</span></span>
<span class="r-in"><span>                  file.name <span class="op">=</span> <span class="st">"MStrat_input"</span>,</span></span>
<span class="r-in"><span>                  folder.path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following MStrat input files created at /var/folders/q_/pk8v8g4n3qq2gcnsvqjc3sd40000gn/T//RtmpVW71TE:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> MStrat_input_data.dat</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> MStrat_input_variable.var</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> MStrat_input_kernel.ker</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by J. Aravind.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.9000.</p>
</div>

    </footer></div>





  </body></html>

